server/jetstream_test.go

```diff
@@ -9953,30 +9953,24 @@ func TestJetStreamPullConsumerMaxWaitingUpdate(t *testing.T) {
 	require_NoError(t, err)
 
 	// Test max waiting requests
-	errch := make(chan error, 2)
+	errch := make(chan error, 1)
 
+	// First is a standard timeout..
 	go func() {
-		_, err := sub.Fetch(1, nats.MaxWait(time.Second))
+		_, err := sub.Fetch(1, nats.MaxWait(300*time.Millisecond))
 		errch <- err
 	}()
 
-	go func() {
-		_, err := sub.Fetch(1, nats.MaxWait(time.Second))
-		errch <- err
-	}()
-
-	var ok bool
-	err = <-errch
-	if strings.Contains(err.Error(), "Exceeded MaxWaiting") {
-		ok = true
-	}
-	err = <-errch
-	if strings.Contains(err.Error(), "Exceeded MaxWaiting") {
-		ok = true
+	// Should be rejected..
```

Comment:
No, we can't rely on the fact that the one in the "go func()" is the first, actually. Do as follow: use same 300ms for both. Check err from inline one: if ErrTimeout, then expect <-errch to be Exceeded. If Exceeded first, expect <-errch to be ErrTimeout. Makes sense?

Reply:
I am glad you said that :wink: Shouldn't be any determinism here.. hopefully should be good now. I appreciate the feedback.